### PR TITLE
docs: add IDE recommendations in README

### LINKED ISSUE #1110: [Site][Discussion]: maintain a VSCode config dir
> Sorry, I need to point out some problems.
> 
> Before you do something, please consider as much as possible whether we really need these improvements, and if you don't know, please create a proposal.
> 
> For example, do we really need a configuration for vscode? Does your configuration work for everyone? I think the answer is obviously no.
> 
> For example, does the addition of jest make sense? Does it cover all use cases that need to be tested?
> 
> There are other problems that I don't want to enumerate.
> 
> Also, I recommend doing one thing per PR, not everything.

I understand your concern about why the `vscode` folder needs to be uploaded. It is because everyone may modify the code of this project, so you should provide a basic configuration of the vscode for this project, which may increase Your code review work, see if anyone has modified this file, but the advantage of this is that everyone will have a default configuration in the project's workspace, which can greatly reduce their configuration of this project time

_Originally posted by @barchielAbyss in https://github.com/apache/apisix-website/issues/1109#issuecomment-1133816229_

### PR DESCRIPTION
**Pull request by barchielAbyss**
Fixes: [#1110](https://github.com/apache/apisix-website/issues/1110)

Changes:
Add `Vscode` recommendations `Extensions` and `Settings` in `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,6 +99,12 @@ apisix-website
     └── docusaurus.config.js
 ```
 
+#### IDE recommendations config
```
Comment (SkyeYoung) [inline README.md]:
```suggestion
#### IDE configuration and plug-in recommendations
```

Maybe better? 🤔Please give your suggestions.😅

File: README.md [hunk 2]
```diff
@@ -99,6 +99,12 @@ apisix-website
...
+
+> Help you configure your IDE quickly
```
Comment (SkyeYoung) [inline README.md]:
```suggestion
> Helps you enhance your development and documentation editing experience.
```

ditto

File: README.md [hunk 3]
```diff
@@ -99,6 +99,12 @@ apisix-website
...
+
+- [VsCode](./website/docs/ide-config.md#vscode)
```
Comment (SkyeYoung) [inline README.md]:
```suggestion
- [VS Code](./website/docs/ide-config.md#vscode)
```

`VsCode` => `VSCode`, `VS Code`, or `Visual Studio Code`

File: website/docs/ide-config.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Vscode
```
Comment (SkyeYoung) [inline website/docs/ide-config.md]:
```suggestion
# VS Code
```
ditto

File: website/docs/ide-config.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+- ESLint
+  - id: "dbaeumer.vscode-eslint"
+  - recommend reason: "Integrates ESLint JavaScript into VS Code."
+- Stylelint
+  - id: "stylelint.vscode-stylelint"
+  - recommend reason: "Official Stylelint extension for Visual Studio Code"
+- GitLens
+  - id: "eamodio.gitlens"
+  - recommend reason: "It helps you to visualize code authorship at a glance via Git blame annotations and CodeLens"
+- markdownlit
```
Comment (SkyeYoung) [inline website/docs/ide-config.md]:
```suggestion
- markdownlint
```

File: website/docs/ide-config.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  - id: "DavidAnson.vscode-markdownlint"
+  - recommend reason: "Markdown linting and style checking for Visual Studio Code"
+
+> Vscode Settings
+
+```json
```
Comment (SkyeYoung) [inline website/docs/ide-config.md]: you can use `js` or `json5` to ignore red highlighting
image

File: website/docs/ide-config.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Vscode
+
+> Vscode Extensions
+
+- ESLint
```
Comment (SkyeYoung) [inline website/docs/ide-config.md:5]: Why not try this way? https://stackoverflow.com/a/52580725/14728656

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

File: website/docs/ide-config.md (hunk 7 above)
Comment (SkyeYoung) [reply website/docs/ide-config.md:5]: Or this way https://tech.serhatteker.com/post/2020-05/vscode-extensions-recommended/

**Comment by SkyeYoung** [comment]
cc @juzhiyuan, pls have a look

File: website/docs/ide-config.md (hunk 7 above)
Comment (barchielAbyss) [reply website/docs/ide-config.md:5]: > Or this way https://tech.serhatteker.com/post/2020-05/vscode-extensions-recommended/

this is my before, 🤣，https://github.com/apache/apisix-website/pull/1109/commits/58fea5dfd567819d85098851ec672b395d8d0936

**Comment by barchielAbyss** [comment]
> cc @juzhiyuan, pls have a look

@juzhiyuan 

Maybe we should discuss the implementation of vscode's recommended configuration again, do you think it's the current one or https://github.com/apache/apisix-website/commit/58fea5dfd567819d85098851ec672b395d8d0936, @SkyeYoung  give me new idea like:

> Why not try this way? https://stackoverflow.com/a/52580725/14728656

File: website/docs/ide-config.md (hunk 7 above)
Comment (SkyeYoung) [reply website/docs/ide-config.md:5]: @barchielAbyss yes, and I mentioned it, but probably didn't make it clear

> What I think is that if we maintain, we must include vscode configuration files and related documents at the same time.
>
> _Originally posted by @SkyeYoung in https://github.com/apache/apisix-website/issues/1111#issuecomment-1133825053_

**Comment by SkyeYoung** [comment]
> > cc @juzhiyuan, pls have a look
> 
> @juzhiyuan
> 
> Maybe we should discuss the implementation of vscode's recommended configuration again, do you think it's the current one or [58fea5d](https://github.com/apache/apisix-website/commit/58fea5dfd567819d85098851ec672b395d8d0936), @SkyeYoung give me new idea like:
> 
> > Why not try this way? [stackoverflow.com/a/52580725/14728656](https://stackoverflow.com/a/52580725/14728656)

In fact, I prefer the second one😅

**Comment by barchielAbyss** [comment]
> In fact, I prefer the second one😅

@SkyeYoung  [juzhiyuan](https://github.com/juzhiyuan) maybe too busy, I'll modify the docs first. As for how to customize the Vsode configuration for the project, you should already know how to do it. :)

File: README.md [hunk 8]
```diff
@@ -99,6 +99,12 @@ apisix-website
     └── docusaurus.config.js
 ```
 
+#### IDE configuration and plug-in recommendations
+
+> Helps you enhance your development and documentation editing experience.
```
Comment (SkyeYoung) [inline README.md]:
```suggestion
> Help you enhance your development and documentation editing experience.
```

File: website/docs/ide-config.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  - id: "stylelint.vscode-stylelint"
+  - recommend reason: "Official Stylelint extension for Visual Studio Code"
+- GitLens
+  - id: "eamodio.gitlens"
+  - recommend reason: "It helps you to visualize code authorship at a glance via Git blame annotations and CodeLens"
+- markdownlint
+  - id: "DavidAnson.vscode-markdownlint"
+  - recommend reason: "Markdown linting and style checking for Visual Studio Code"
+
+> Vscode Settings
```
Comment (SkyeYoung) [inline website/docs/ide-config.md]:
```suggestion
> VS Code Settings
```

File: website/docs/ide-config.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
+<h2 id="vscode">VS Code</h2>
+
+> Vscode Extensions
```
Comment (SkyeYoung) [inline website/docs/ide-config.md]:
```suggestion
> VS Code Extensions
```

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

**Comment by SkyeYoung** [comment]
cc @juzhiyuan

**Comment by SkyeYoung** [comment]
Hi, pls add `Fixes: #1110` to link to origin issue.

File: README.md [hunk 11]
```diff
@@ -99,6 +99,12 @@ apisix-website
     └── docusaurus.config.js
 ```
 
+#### IDE configuration and plug-in recommendations
```
Comment (juzhiyuan) [inline README.md]:
```suggestion
#### IDE configuration and plugin recommendations
```

**Review by juzhiyuan — COMMENTED** [review]
🤔 I'm not sure, but why not directly put a `.vscode` configuration under the repo?

File: README.md (hunk 11 above)
Comment (barchielAbyss) [reply README.md]: Let's add both forms, the document is to declare what helps to configure, and .config is the action of directly configuring :)

File: website/docs/ide-config.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  "files.eol": "\n",
+  "files.insertFinalNewline": true,
+  "files.trimFinalNewlines": true,
+
+  // ===
+  // Files
+  // ===
+
+  "files.exclude": {
+    "**/dist": true
```
Comment (SkyeYoung) [inline website/docs/ide-config.md:39]: Does this rule cover all the files that need to be excluded?

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

File: website/docs/ide-config.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+  - id: "stylelint.vscode-stylelint"
+  - recommend reason: "Official Stylelint extension for Visual Studio Code"
+- GitLens
+  - id: "eamodio.gitlens"
+  - recommend reason: "It helps you to visualize code authorship at a glance via Git blame annotations and CodeLens"
+- markdownlint
+  - id: "DavidAnson.vscode-markdownlint"
+  - recommend reason: "Markdown linting and style checking for Visual Studio Code"
+
+> VS Code Settings
```
Comment (SkyeYoung) [inline website/docs/ide-config.md:18]: The other thing is, how do we make sure that the configuration in the `.vscode` folder and here can be maintained at the same time? I would suggest adding a "Tip" here

File: website/docs/ide-config.md (hunk 12 above)
Comment (barchielAbyss) [reply website/docs/ide-config.md:39]: Configure glob patterns for excluding files and folders in fulltext searches and quick open. Inherits all glob patterns from the `files.exclude` setting.

https://code.visualstudio.com/docs/getstarted/settings

Comment (SkyeYoung) [reply website/docs/ide-config.md:39]: @barchielAbyss I mean, we already have .gitignore, do we really need this configuration? And we don't seem to be generating any folders named dist?

**Comment by SkyeYoung** [comment]
@barchielAbyss Any update here? I think we can end this PR soon. 🧠

**Pull request closed without merging**